metau-capstone/ScheduleViewController.m

```diff
@@ -39,4 +42,74 @@ - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
 }
 */
 
+- (nonnull __kindof UICollectionViewCell *)collectionView:(nonnull UICollectionView *)collectionView cellForItemAtIndexPath:(nonnull NSIndexPath *)indexPath {
+    NSNumber *cellNum = @(indexPath.row+1);
+    ScheduleCell *cell = [self.scheduleCollection dequeueReusableCellWithReuseIdentifier:@"ScheduleCollectionCell" forIndexPath:indexPath];
+    cell.dayNum.text = [NSString stringWithFormat:@"%ld", indexPath.row+1];
+    cell.layer.borderColor = [UIColor blackColor].CGColor;
+    cell.layer.borderWidth = 1;
+    
+    // Query for user's day
+    PFQuery *queryForDay = [PFUser query];
+    PFUser *const user = [PFUser currentUser];
+    [queryForDay getObjectInBackgroundWithId:user.objectId block:^(PFObject * _Nullable object, NSError * _Nullable error) {
+        if (!error) {
+            // Highlight current day
+            if ([object[@"userDay"] isEqualToNumber:cellNum]) {
+                cell.dayNum.backgroundColor = [UIColor yellowColor];
+            } else {
+                cell.dayNum.backgroundColor = [UIColor clearColor];
+            }
+        }
+    }];
+    
+    //Query for the day's levels
+    PFQuery *queryForLevels = [PFQuery queryWithClassName:@"Schedule"];
+    [queryForLevels whereKey:@"dayNum" equalTo:cellNum];
+    [queryForLevels findObjectsInBackgroundWithBlock:^(NSArray * _Nullable objects, NSError * _Nullable error) {
+        if (!error) {
+            for (Schedule *object in objects) {
+                NSArray *arrayOfLevels = object.arrayOfLevels;
+                NSString *levelsText = @"";
+                for (int i = 0; i < arrayOfLevels.count; i++) {
+                    if (i == 0) {
+                        levelsText = [levelsText stringByAppendingFormat:@"Level %@", arrayOfLevels[i]];
+                    } else {
+                        levelsText = [levelsText stringByAppendingFormat:@"\rLevel %@", arrayOfLevels[i]];
+                    }
+                }
+                cell.levelsLabel.text = levelsText;
+            }
+        }
+    }];
+    return cell;
+}
+
+- (NSInteger)collectionView:(nonnull UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section {
+    return 64;
```

Comment:
Should this be a dynamic number which is the number of actual cards

Reply:
This is the number of days in the Leitner Schedule. So there is a 64 day cycle, which is why I have this number hard coded. I could call the number of Schedule items but since I am currently storing that in Parse I would have to call my database.